plugins/experimental/maxmind_acl/maxmind_acl.cc

```diff
@@ -0,0 +1,117 @@
+/*
+  Licensed to the Apache Software Foundation (ASF) under one
+  or more contributor license agreements.  See the NOTICE file
+  distributed with this work for additional information
+  regarding copyright ownership.  The ASF licenses this file
+  to you under the Apache License, Version 2.0 (the
+  "License"); you may not use this file except in compliance
+  with the License.  You may obtain a copy of the License at
+
+  http://www.apache.org/licenses/LICENSE-2.0
+
+  Unless required by applicable law or agreed to in writing, software
+  distributed under the License is distributed on an "AS IS" BASIS,
+  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  See the License for the specific language governing permissions and
+  limitations under the License.
+*/
+
+#include "mmdb.h"
+
+static int
+config_handler(TSCont cont, TSEvent event, void *edata)
+{
+  TSMutex mutex;
+
+  mutex = TSContMutexGet(cont);
+  TSMutexLock(mutex);
+
+  TSDebug(PLUGIN_NAME, "In config Handler");
+  Acl *a = static_cast<Acl *>(TSContDataGet(cont));
+
+  // strdup for const string return
+  char *config = strdup(a->get_state()->config_file.c_str());
+  a->init(config);
```

Comment:
Why not change the type of the file spec parameter to init() to char const * so this could just be:
a->init(a->get_state()->config_file.c_str());

Reply:
thanks, Ill take a look